queue/src/fiskaltrust.Middleware.Localization.QueueIT/RequestCommands/PosReceiptCommand.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using fiskaltrust.ifPOS.v1;
+using fiskaltrust.storage.V0;
+using fiskaltrust.Middleware.Contracts.RequestCommands;
+using fiskaltrust.ifPOS.v1.it;
+using System.Linq;
+using fiskaltrust.Middleware.Localization.QueueIT.Extensions;
+using fiskaltrust.Middleware.Localization.QueueIT.Factories;
+using fiskaltrust.Middleware.Localization.QueueIT.Services;
+using fiskaltrust.Middleware.Contracts.Extensions;
+
+namespace fiskaltrust.Middleware.Localization.QueueIT.RequestCommands
+{
+    public class PosReceiptCommand : RequestCommandIT
+    {
+        private readonly SignatureItemFactoryIT _signatureItemFactoryIT;
+        private readonly IITSSCD _client;
+
+        public PosReceiptCommand(IITSSCDProvider itIsscdProvider, SignatureItemFactoryIT signatureItemFactoryIT)
+        {
+            _client = itIsscdProvider.Instance;
+            _signatureItemFactoryIT = signatureItemFactoryIT;
+        }
+        public override async Task<RequestCommandResponse> ExecuteAsync(ftQueue queue, ReceiptRequest request, ftQueueItem queueItem)
+        {
+            var receiptResponse = CreateReceiptResponse(queue, request, queueItem, CountryBaseState);
+
+            FiscalReceiptResponse response;
+            if (!request.IsVoid())
+            {
+                var fiscalReceiptinvoice = CreateInvoice(request);
+                response = await _client.FiscalReceiptInvoiceAsync(fiscalReceiptinvoice).ConfigureAwait(false);
+            }
+            else
+            {
+                var fiscalReceiptRefund = CreateRefund(request);
+                response = await _client.FiscalReceiptRefundAsync(fiscalReceiptRefund).ConfigureAwait(false);
+            }
+            receiptResponse.ftSignatures = _signatureItemFactoryIT.CreatePosReceiptSignatures(response);
+
+            if (!response.Success)
+            {
+                throw new Exception(response.ErrorInfo);
```

Comment:
Let's maybe add some more information here, or (better) use a dedicated exception class for this.